src/libxively/sft/xi_sft_revision.c

```diff
@@ -77,15 +77,15 @@ static xi_state_t xi_sft_revision_read_string_from_file( const char* const resou
 
     XI_CHECK_STATE( state );
 
-    xi_bsp_io_fs_close( resource_handle );
-
     /* allocate output buffer then copy revision buffer into output buffer */
     XI_ALLOC_BUFFER_AT( char, *string_read_out, buffer_size + 1, state );
 
     memcpy( *string_read_out, buffer, buffer_size );
 
 err_handling:
 
+    xi_bsp_io_fs_close( resource_handle );
+
```

Comment:
HINT: this was a bugfix for @Palantir555 ESP32 platform, but probably only worked for POSIX by accident. Since the read buffer was deallocated by the xi_bsp_io_fs_close and used afterwards. ESP32 changed the first bytes of the deallocated memory area but POSIX didn't thus POSIX worked well. But yeah, probably this is the undefined behaviour situation. Who knows what can happen.

Reply:
I've also included this fix in my branch. I fs_close() outside of the err_handling subroutine, though. Should I remove it from my branch to alleviate conflicts?

Reply:
well that's a son of a b*tch.